specification/resources/resource-manager/Microsoft.Management/preview/2018-01-01-preview/management.json

```diff
@@ -0,0 +1,1266 @@
+          "type": "string",
+          "description": "The fully qualified ID for the parent management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000"
+        }
+      }
+    },
+    "EntityHierarchyItem": {
+      "description": "The management group details for the hierarchy view.",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified ID for the management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the resource.  For example, /providers/Microsoft.Management/managementGroups",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the management group. For example, 00000000-0000-0000-0000-000000000000",
+          "readOnly": true
+        },
+        "properties": {
+          "title": "Properties",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/EntityHierarchyItemProperties"
+        }
+      }
+    },
+    "EntityHierarchyItemProperties": {
+      "description": "The generic properties of a management group.",
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The friendly name of the management group."
+        },
+        "permissions": {
+          "title": "Permissions",
+          "$ref": "#/definitions/Permissions"
+        },
+        "children": {
+          "type": "array",
+          "description": "The list of children.",
+          "x-nullable": true,
+          "items": {
+            "$ref": "#/definitions/EntityHierarchyItem"
+          }
+        }
+      }
+    },
+    "PatchManagementGroupRequest": {
+      "description": "Management group patch parameters.",
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The friendly name of the management group.",
+          "x-nullable": true
+        },
+        "parentId": {
+          "type": "string",
+          "description": "(Optional) The fully qualified ID for the parent management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000",
+          "x-nullable": true
+        }
+      }
+    },
+    "CreateManagementGroupRequest": {
+      "description": "Management group creation parameters.",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified ID for the management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000",
+          "readOnly": true
```

Comment:
>readOnly [](start = 11, length = 8)

Why would there be readonly parameters in a model that is used to create a resource?

Reply:
Because we only actually take the display name field and the parent id.  This was done as our review wanted the put structure to more closely match the get structure.

Reply:
The API review board you mean?

Reply:
Yes - sorry, was typing from my phone.

Reply:
okok :)